autobot/telegram/parser.py

```diff
@@ -697,3 +697,169 @@ def _parse_sticker (self,key:str,val:dict) -> Sticker|StickerSet:
+                case "title"|"address"|"foursquare_id":
+                    setattr(venue_obj,k,v)
+                case _:
+                    setattr(venue_obj,k,v)
+        return venue_obj
+
+    def _parse_photosize(self,key:str,val:dict) -> PhotoSize:
+        photosize_obj = PhotoSize()
+        for k,v in val.items():
+            match k:
+                case "file_size":
+                    setattr(photosize_obj,k,int(v))
+                case _:
+                    setattr(photosize_obj,k,v)
+        return photosize_obj
+
+    def _parse_location(self,key:str,val:dict) -> Location:
+        location_obj = Location()
+        for k,v in val.items():
+            match k:
+                case "longitude"|"latitude":
+                    setattr(location_obj,k,v)
+                case _:
+                    setattr(location_obj,k,v)
+        return location_obj
+
+
+    def _parse_contact(self,key:str,val:dict) -> Contact:
+        contact_obj = Contact()
+        for k,v in val.items():
+            match k:
+                case "phone_number"|"first_name"|"last_name":
+                    setattr(contact_obj,k,v)
+                case _:
+                    setattr(contact_obj,k,v)
+        return contact_obj
+
+    def _parse_proximityalerttriggered(self,key:str,val:dict) -> ProximityAlertTriggered:
+        proximityalerttriggered_obj = ProximityAlertTriggered()
+        for k,v in val.items():
+            match k:
+                case "traveler"|"watcher"|"distance":
+                    setattr(proximityalerttriggered_obj,k,v)
+                case _:
+                    setattr(proximityalerttriggered_obj,k,v)
+        return proximityalerttriggered_obj
+
+    def _parse_login_url(self,key:str,val:dict) -> LoginUrl:
+        loginurl_obj = LoginUrl()
+        for k,v in val.items():
+            match k:
+                case "url"|"forward_text"|"bot_username"|"request_write_access":
+                    setattr(loginurl_obj,k,v)
+                case _:
+                    setattr(loginurl_obj,k,v)
+        return loginurl_obj
+
+
+    def _parse_maskposition(self,key:str,val:dict) -> MaskPosition:
+        maskposition_obj = MaskPosition()
+        for k,v in val.items():
+            match k:
+                case "point"|"x_shift"|"y_shift"|"scale":
+                    setattr(maskposition_obj,k,v)
+                case _:
+                    setattr(maskposition_obj,k,v)
+        return maskposition_obj
+
+    def _parse_inlinekeyboardbutton(self,key:str,val:dict) -> InlineKeyboardButton:
+        inlinekeyboardbutton_obj = InlineKeyboardButton()
+        for k,v in val.items():
+            match k:
+                case "text"|"url"|"callback_data"|"switch_inline_query"|"switch_inline_query_current_chat"|"callback_game"|"pay":
+                    setattr(inlinekeyboardbutton_obj,k,v)
+                case _:
+                    setattr(inlinekeyboardbutton_obj,k,v)
+        return inlinekeyboardbutton_obj
```

Comment:
In cases like this where methods do not delegate parsing to any other methods, that is, all attributes can be fully parsed by that single method, then there's no need to have `match case` statements. Instead the the for loop handle `setattr` entirely and return the result